### PR TITLE
Read only view in questions/forms are too light.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Issue #917 

**Description**

-  Override default theme text color to improve contrast of disabled text. ` android:textColor `
-  Override default theme box stroke color to improve contrast of disabled widget. `boxStrokeColor`

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one:  Feature 

**Screenshots (if applicable)**

 Before implementation 
 _Edit text with Mark value_
Screen Shot 2021-11-24 at 3 21 36 PM

 After implementation 
 _Edit text with Mark value_
Screen Shot 2021-11-24 at 3 24 29 PM

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by santosh-pingle** [comment]
@jing could you please confirm proposed implementation?

File: datacapture/src/main/res/color/box_stroke_color.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+  <item android:color="?attr/colorPrimary" android:state_focused="true" />
+  <item
+        android:alpha="0.87"
+        android:color="?attr/colorOnSurface"
+        android:state_hovered="true"
+    />
+  <item android:color="#9E9E9E" android:state_enabled="false" />
+  <item android:color="#5F6368" />
+</selector>
```
Comment (jingtang10) [inline datacapture/src/main/res/color/box_stroke_color.xml:11]: assuming you modified this from the android source? can you link to the original file and comment on the diff?

File: datacapture/src/main/res/color/text_color.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+  <item android:color="#9E9E9E" android:state_enabled="false" />
+  <item android:color="#5F6368" />
+</selector>
```
Comment (jingtang10) [inline datacapture/src/main/res/color/text_color.xml:5]: assuming you modified this from the android source? can you link to the original file and comment on the diff?

File: datacapture/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -41,5 +41,35 @@
...
+        <item
+            name="editTextStyle"
+        >@style/Questionnaire.Widget.MaterialComponents.TextInputEditText.OutlinedBox</item>
+    </style>
+
+    <style
+        name="Questionnaire.Widget.MaterialComponents.TextInputEditText.OutlinedBox"
+        parent="Widget.MaterialComponents.TextInputEditText.OutlinedBox"
+    >
+        <item name="android:textColor">@color/text_color</item>
```
Comment (jingtang10) [inline datacapture/src/main/res/values/styles.xml:73]: Add comment:

`Override default theme text color to improve contrast of disabled text.`

File: datacapture/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -41,5 +41,35 @@
...
+    </style>
+
+    <style
+        name="Questionnaire.Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+        parent="Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+    >
+        <item name="materialThemeOverlay">
+            @style/Questionnaire.Widget.ThemeOverlay.MaterialComponents.TextInputEditText
+        </item>
+        <item name="boxStrokeColor">@color/box_stroke_color</item>
```
Comment (jingtang10) [inline datacapture/src/main/res/values/styles.xml:57]: Add comment:

`Override default theme box stroke color to improve contrast of disabled widget.`

**Review by jingtang10 — CHANGES_REQUESTED** [review]

**Comment by jingtang10** [comment]
please also update the PR name and description

**Comment by santosh-pingle** [comment]
Updating code changes as per review comments.

File: datacapture/src/main/res/color/text_color.xml (hunk 2 above)
Comment (santosh-pingle) [reply datacapture/src/main/res/color/text_color.xml:5]: it is newly created color state list resource.

**Comment by jingtang10** [comment]
please mark as ready to review if ready. thanks!

File: datacapture/src/main/res/color/box_stroke_color.xml (hunk 1 above)
Comment (santosh-pingle) [reply datacapture/src/main/res/color/box_stroke_color.xml]: I have update comment with file name, please check once.

**Comment by santosh-pingle** [comment]
@jingtang10 
Implemented contrast change is only visible to questionnaire edit text view (`QuestionnaireItemEditTextViewHolderFactory`) and not for others. Do we also need to apply it to other questionnaire views?

**Comment by jingtang10** [comment]
> @jingtang10 Implemented contrast change is only visible to questionnaire edit text view (`QuestionnaireItemEditTextViewHolderFactory`) and not for others. Do we need to apply it to other questionnaire views?

yeah you do. anything with the edit text you'll need to apply the stroke color, anything with text you'll probably need to apply the text color.

**Comment by jingtang10** [comment]
have you also tested this with dark theme? please attach some screenshots. thanks!

**Comment by jingtang10** [comment]
also please attach before/after screenshots with other widgets if you do them in the same PR.

**Comment by santosh-pingle** [comment]
> have you also tested this with dark theme? please attach some screenshots. thanks!

looking into it.

**Comment by santosh-pingle** [comment]
> > have you also tested this with dark theme? please attach some screenshots. thanks!
> 
> looking into it.

@shelaghm 
when `#9E9E9E` get applied to the disabled text and container it works as expected in light theme (please refer earlier screenshots) but when it switch to the dark theme then contrast/opacity is too high. 
To fix this issue, it is required to use alpha value and theme attribute to create color state list resource. 
As you mentioned **38% opacity** in earlier comments I assumed **alpha value is .38** as range is 0 to 1, now it works  in both themes, but if you think alpha value can be greater than .38 please let me know.

Light theme : read only view with alpha .38

light theme alpha  38

Dark theme : readonly view with alpha .38
dark theme with alpha  38

**Comment by santosh-pingle** [comment]
autocomplete textview, light theme, alpha .38
light theme autocomplete  38

autocomplete textview, dark theme, alpha .38

dark theme autocomplete  38

**Comment by santosh-pingle** [comment]
Q Hep B y / n readonly view `QuestionnaireItemBooleanTypePickerViewHolderFactory`, alpha .38, light theme
light boolean  38

Q Hep B y / n readonly view `QuestionnaireItemBooleanTypePickerViewHolderFactory`, alpha .38, dark theme
dark boolean  38

**Comment by santosh-pingle** [comment]
Q. 1st dose 2nd dose readonly view `QuestionnaireItemDateTimePickerViewHolderFactory`, light theme, alpha .38
 
datetime picker light alpha  38


Q. 1st dose 2nd dose readonly view `QuestionnaireItemDateTimePickerViewHolderFactory`, dark theme, alpha .38

datetimepicker dark alpha  38

**Comment by santosh-pingle** [comment]
Q.1 & Q.3 readonly view `QuestionnaireItemDialogSelectViewHolderFactory`, light theme, alpha .38
light drop down dialog select alpha  38


Q.1 & Q.3 readonly view `QuestionnaireItemDialogSelectViewHolderFactory`, dark theme, alpha .38
dark drop down dialog select alpha  38

**Comment by santosh-pingle** [comment]
@jingtang10
Can you please it? Thanks!

**Comment by santosh-pingle** [comment]
readonly view `QuestionnaireItemDropDownViewHolderFactory` alpha .38

Screen Shot 2021-12-03 at 4 47 41 PM


Screen Shot 2021-12-03 at 4 48 26 PM

**Comment by santosh-pingle** [comment]
readonly view `QuestionnaireItemCheckBoxGroupViewHolderFactory` alpha .38
Screen Shot 2021-12-03 at 4 47 20 PM


Screen Shot 2021-12-03 at 4 46 44 PM

**Review by jingtang10 — APPROVED** [review]

**Comment by anita-vadverao** [comment]
Retested "Readonly" changes by adding "readOnly" :  true" attribute in Questionnaire JSON and it is properly visible after adding contrast to text. Working as expected.

**Pull request merged**